views/start.dt

```diff
@@ -0,0 +1,98 @@
+extends layout
+
+block title
+	- import dubregistry.viewutils;
+	- import dubregistry.web;
+	- import vibe.data.json;
+	- import std.algorithm.iteration : map;
+	- auto title = "Find, Use and Share DUB Packages";
+	script(type="application/javascript", src="scripts/home.js")
+	script(type="application/javascript").
+		window.categories = #{serializeToJson(info.categories).toString()};
+
+block body
+	- auto mirror = req.params.get("mirror", "");
+	- if (mirror.length)
+		p This is a mirror of #[a(href=mirror)= mirror].
+	- else
+		p Welcome to DUB, the D package registry.
+
+	- string makeHomeURL(string sort) {
+	-   return "?sort=" ~ sort;
+	- }
+
+	p Total #{info.packageCount} packages found.
+
+		form#category-form(method="GET", action="")
+			p Select category:
+				select#category(name="category", size="1", onChange='document.getElementById("category-form").submit()')
+					- void outputCat(Category cat)
+						- if (!cat)
+							option(value="") All packages
+						- else
+							option(value=cat.name)= cat.indentedDescription
+						- if (!cat || cat.subCategories.length)
+							- foreach (c; cat ? cat.subCategories : info.categories)
+								- outputCat(c);
+					- outputCat(null);
+				button#category-submit(type="submit") Update
+		#category-dynamic-form(style="display: none")
+			p Select category:
+				- foreach (i; 0 .. 6)
+					select(id="categories_#{i}", name="categories_#{i}", onChange='setCategoryFromSelector(#{i})')
+			:javascript
+				setupCategoryForm();
+
+		- void listPackages(P)(P packages) {
+			ul
+				- foreach (pl; packages)
+					- if( pl["versions"].length )
+						- auto vidx = getBestVersionIndex(pl["versions"].get!(Json[]).map!(v => v["version"].opt!string));
+						- auto p = pl["versions"][vidx];
+						- auto desc = p["description"].opt!string;
+						- auto ver = p["version"].opt!string();
+						li
+							.front-package-logo
+								- import std.string : startsWith;
+								- string image, imageDescription;
+								- bool isApplication;
+								- if (pl["categories"].length)
+									- foreach (category; pl["categories"])
+										- if (category.get!string.startsWith("application"))
+											- isApplication = true;
+											- break;
+									- if (auto pc = pl["categories"][0].get!string in info.categoryMap)
+										- image = pc.imageName;
+										- imageDescription = pc.imageDescription;
+								- if (!image.length) image = "unknown";
+								- if (!imageDescription.length) imageDescription = "unknown";
+								span(class="category-icon icon-category-#{image} #{isApplication ? `application` : ``}", title=imageDescription)
```

Comment:
Should the logo be bigger? (maybe a good idea - especially with the new logo upload functionality)